crates/chain/tests/test_indexed_tx_graph.rs

```diff
@@ -1,19 +1,331 @@
+        tx_name: "tx2",
+        outputs: &[TxOutTemplate {
+            value: 30000,
+            owned_spk: Some((TestKeychain::Internal, 0)),
+        }],
+        anchors: &[ConfirmationHeightAnchor {
+            anchor_block: BlockId {
+                height: 1,
+                hash: *local_chain.blocks().get(&1).unwrap(),
+            },
+            confirmation_height: 1,
+        }],
+        last_seen: None,
+        ..Default::default()
+    };
+
+    // tx3 spends tx2 and gives a change back in trusted keychain. Confirmed at Block 2.
+    let tx3: TxTemplate<'_, _, ConfirmationHeightAnchor> = TxTemplate {
+        tx_name: "tx3",
+        inputs: &[TxInTemplate::PrevTx("tx2", 0)],
+        outputs: &[TxOutTemplate {
+            value: 10000,
+            owned_spk: Some((TestKeychain::External, 1)),
+        }],
+        anchors: &[ConfirmationHeightAnchor {
+            anchor_block: BlockId {
+                height: 2,
+                hash: *local_chain.blocks().get(&2).unwrap(),
+            },
+            confirmation_height: 2,
+        }],
+        last_seen: None,
+    };
+
+    // tx4 is an external transaction receiving at untrusted keychain, unconfirmed.
+    let tx4: TxTemplate<'_, _, ConfirmationHeightAnchor> = TxTemplate {
+        tx_name: "tx4",
+        outputs: &[TxOutTemplate {
+            value: 20000,
+            owned_spk: Some((TestKeychain::Internal, 1)),
+        }],
+        last_seen: Some(100),
+        ..Default::default()
+    };
+
+    // tx5 is spending tx3 and receiving change at trusted keychain, unconfirmed.
+    let tx5: TxTemplate<'_, _, ConfirmationHeightAnchor> = TxTemplate {
+        tx_name: "tx5",
+        outputs: &[TxOutTemplate {
+            value: 15000,
+            owned_spk: Some((TestKeychain::External, 2)),
+        }],
+        last_seen: Some(100),
+        ..Default::default()
+    };
+
+    // tx6 is an irrelevant transaction confirmed at Block 3.
+    let tx6 = TxTemplate {
+        tx_name: "tx6",
+        anchors: &[ConfirmationHeightAnchor {
+            anchor_block: BlockId {
+                height: 3,
+                hash: *local_chain.blocks().get(&3).unwrap(),
+            },
+            confirmation_height: 3,
+        }],
+        last_seen: None,
+        ..Default::default()
+    };
+
+    let scenarios = [Scenario {
+        name: "test_list_owned_txouts",
+        tx_templates: &[tx1, tx2, tx3, tx4, tx5, tx6],
+        exp_chain_txs: HashSet::from(["tx1", "tx2", "tx3", "tx4", "tx5"]),
+        exp_unspents: HashSet::from([("tx1", 0), ("tx3", 0), ("tx4", 0), ("tx5", 0)]),
+        exp_balance: Balance {
+            immature: 0,              // coinbase matured
+            trusted_pending: 15000,   // tx5
+            untrusted_pending: 20000, // tx4
+            confirmed: 80000,         // tx1 + tx3
+        },
+    }];
+
+    // A helper lambda to extract and filter data from the graph.
+    let fetch =
```

Comment:
We don't need a separate closure here since we are only calling it in one place, but I guess this is a placeholder test so it'll be removed anyway?